packages/picasso-codemod/src/v22.0.0/page-sidebar/page-sidebar.ts

```diff
@@ -0,0 +1,91 @@
+import core, {
+  Transform,
+  JSCodeshift,
+  Collection,
+  ImportDeclaration
+} from 'jscodeshift'
+
+import {
+  addImport,
+  findComponents,
+  findSpecifierForImport,
+  isImportFor,
+  isSpecifierFor
+} from '../../utils'
+import { PackageMember } from '../../utils/types'
+
+const PICASSO_PACKAGE = '@toptal/picasso'
+
+const OLD_PICASSO_SIDEBAR: PackageMember = {
+  packageName: PICASSO_PACKAGE,
+  exportedName: 'Sidebar'
+}
+
+const PICASSO_PAGE: PackageMember = {
+  packageName: PICASSO_PACKAGE,
+  exportedName: 'Page'
+}
+
+const transformImports = (root: Collection) => {
+  const pageIdentifier = findSpecifierForImport(PICASSO_PAGE, root)
+
+  if (pageIdentifier == null) {
+    addImport(PICASSO_PAGE, root)
+  }
+
+  root
+    .find(core.ImportDeclaration, decl =>
+      isImportFor(OLD_PICASSO_SIDEBAR, decl)
+    )
```

Comment:
About the usage of `isImportFor`, since we are using `decl` twice and the function is actually really simple, I have an idea like this:

```suggestion
// as an independent definition
const isImportForOldPicassoSidebar = generateIsImportFor(OLD_PICASSO_SIDEBAR)
    //-----

    // then inside transform function
    .find(core.ImportDeclaration, isImportForOldPicassoSidebar)
    )
```
I think it would be more helpful to get the actual idea of the codemod when we apply this to other utils as well, WDYT?

Reply:
This was actually my first implementation, all of the util functions we are actually [curried](https://javascript.info/currying-partials) when I wrote them. And even though some places looked neater I preferred to remove them and go simpler as not a lot of devs are familiar with the concept.

The first implementation was something like:
```suggestion
    .find(core.ImportDeclaration, isImportFor(OLD_PICASSO_SIDEBAR))
```

Maybe we could add lodash as a dependency to the project, this would make auto-currying a lot more approachable, and optional